Ryujinx.HLE/HOS/Services/Ro/IRoInterface.cs

```diff
@@ -187,77 +193,118 @@ public ResultCode ParseNro(out NroInfo res, ServiceCtx context, ulong nroAddress
             return ResultCode.Success;
         }
 
-        private ResultCode MapNro(ServiceCtx context, NroInfo info, out ulong nroMappedAddress)
+        private ResultCode MapNro(KProcess process, NroInfo info, out ulong nroMappedAddress)
         {
-            nroMappedAddress = 0;
+            KMemoryManager memMgr = process.MemoryManager;
 
-            KMemoryManager memMgr = context.Process.MemoryManager;
+            int retryCount = 0;
 
-            ulong targetAddress = memMgr.GetAddrSpaceBaseAddr();
+            nroMappedAddress = 0;
 
-            while (true)
+            while (retryCount++ < MaxMapRetries)
             {
-                if (targetAddress + info.TotalSize >= memMgr.AddrSpaceEnd)
+                ResultCode result = MapCodeMemoryInProcess(process, out nroMappedAddress, info.NroAddress, info.NroSize);
+
+                if (result != ResultCode.Success)
                 {
-                    return ResultCode.InvalidMemoryState;
+                    return result;
                 }
 
-                KMemoryInfo memInfo = memMgr.QueryMemory(targetAddress);
-
-                if (memInfo.State == MemoryState.Unmapped && memInfo.Size >= info.TotalSize)
+                if (info.BssSize > 0)
                 {
-                    if (!memMgr.InsideHeapRegion (targetAddress, info.TotalSize) &&
-                        !memMgr.InsideAliasRegion(targetAddress, info.TotalSize))
+                    KernelResult bssMappingResult = memMgr.MapProcessCodeMemory(nroMappedAddress + info.NroSize, info.BssAddress, info.BssSize);
+
+                    if (bssMappingResult == KernelResult.InvalidMemState)
                     {
-                        break;
+                        memMgr.UnmapProcessCodeMemory(nroMappedAddress + info.NroSize, info.BssAddress, info.BssSize);
+                        memMgr.UnmapProcessCodeMemory(nroMappedAddress, info.NroAddress, info.NroSize);
```

Comment:
```suggestion
                        memMgr.UnmapProcessCodeMemory(nroMappedAddress + info.NroSize, info.BssAddress, info.BssSize);
                        memMgr.UnmapProcessCodeMemory(nroMappedAddress,                info.NroAddress, info.NroSize);
```
Maybe you could align args here to be more readable

Reply:
Imo, this makes it less readable.

Reply:
As I said, it's just a suggestion...

Reply:
yeah I will not apply this one